connectors/src/connectors/webcrawler/temporal/activities.ts

```diff
@@ -127,29 +131,62 @@ export async function crawlWebsiteByConnectorId(connectorId: ModelId) {
           Context.current().heartbeat({
             type: "upserting",
           });
-          await upsertToDatasource({
-            dataSourceConfig,
-            documentId: documentId,
-            documentContent: {
-              prefix: pageTitle,
-              content: extracted,
-              sections: [],
-            },
-            documentUrl: request.url,
-            timestampMs: new Date().getTime(),
-            tags: [`title:${pageTitle.substring(0, 300)}`],
-            parents: getParentsForPage(request.url, false),
-            upsertContext: {
-              sync_type: "batch",
-            },
-          });
+
+          try {
+            if (
+              extracted.length > 0 &&
+              extracted.length <= MAX_DOCUMENT_TXT_LEN
+            ) {
+              await upsertToDatasource({
+                dataSourceConfig,
+                documentId: documentId,
+                documentContent: {
+                  prefix: pageTitle,
```

Comment:
You want to add a \n after the title otherwise there won’t be any return line. Maybe prefix with TITLE:

Reply:
Maybe use the same capping to 300 character as well and I would also try to add the URL here as well since we know title are not awesome and having the url seems super useful ?